File: app/src/main/java/com/example/minilist/edit/EditActivityPresenter.java
package com.example.minilist.edit;

import java.io.BufferedWriter;
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileWriter;
import java.io.IOException;
import java.util.Scanner;

import static com.example.minilist.main.MainPresenter.FILE_NAME;

public class EditActivityPresenter {

    private final EditActivity wireframe;

    EditActivityPresenter(EditActivity wireframe) {
        this.wireframe = wireframe;
        File source = new File(wireframe.getFilesDir(), FILE_NAME);
        if (source.exists()) {
            try (Scanner in = new Scanner(source)) {
                this.wireframe.setText(in.useDelimiter("\\Z").next());
            } catch (FileNotFoundException exception) {
                exception.printStackTrace();
            }
        }
    }

    public void saveText(String text) {
        File file = new File(wireframe.getFilesDir(), FILE_NAME);
        if (text.isEmpty()) {
            //del
            file.delete();
        } else {
            //save
            try (BufferedWriter out = new BufferedWriter(new FileWriter(file))) {
                out.write(text);
            } catch (IOException ioe) {
                ioe.printStackTrace();
            }
        }
    }


}

File: app/src/main/java/com/example/minilist/edit/EditActivityPresenter.kt
package com.example.minilist.edit

import com.example.minilist.Shared
import com.example.minilist.main.MainPresenter
import java.io.*
import java.util.*

class EditActivityPresenter internal constructor(private val wireframe: EditWireframe) {

    init {
        val source = File(wireframe.getFilesDir(), Shared.FILE_NAME)
        if (source.exists()) {
            try {
                wireframe.setText(source.readText())
            } catch (exception: FileNotFoundException) {
                exception.printStackTrace()
            }
        }
    }

    fun saveText(text: String) {
        val file = File(wireframe.getFilesDir(), Shared.FILE_NAME)
        if (text.isEmpty()) {
            //del
            file.delete()
        } else {
            //save
            try {
                file.writeText(text)
            } catch (ioe: IOException) {
                ioe.printStackTrace()
            }
        }
    }
}